doc/content/xapi/migration_overview.md

```diff
@@ -0,0 +1,87 @@
+---
+
+# How the migration request goes through Xen API
+
+## Overview
+
+In this document we will utilize the `VM.pool_migrate` request to illustrate
+the interaction between various components within the XAPI toolstack during
+migration. However this schema can be applied to other requests as well.
+
+Not all parts of the Xapi toolstack are shown here as not all are involved in
+the migration process. For instance you won't see the ***squeezed***
+nor ***mpathalert*** two daemons that belong to the toolstack but don't
+participate in the migration of a VM.
+
+## Anatomy of a VM migration
+
+- Migration is initiated by a Xapi client that sends `VM.pool_migrate`, an RPC XML request.
+- The Xen API server handles this request and dispatches it to the server.
+- The server is generated using `XAPI IDL` and requests are wrapped whitin a
+context, either to be forwared to a host or executed locally. Broadly, the
+context follows RBAC rules. The executed function is related to the message of
+the request (refer to [XenAPI Reference](https://xapi-project.github.io/xen-api/)).
+- In the case of the migration you can refer to *ocaml/idl/datamodel_vm.ml*.
+- The server will dispatch the operation to server helpers, executing the
+operation synchronously or asynchronously and returning the RPC answer.
+- Operations can then be forwarded to another host in the pool or executed
+locally.
+- When executed locally the high-level migration operation is send to the
+*Xenopsd daemon* by posting a message on the message switch.
+*Xenopsd daemon* will split this operation into several *atomic* operations
+that will be run by the *xenopsd backend* in XAPI. The steps involve:
+    - Splitting operations into multiple backend operations.
+    - Posting this operation as an RPC to *xenopsd server*.
+    - The *xenopsd server* then pushes operations onto the message switch.
+    - This allows the *xenopsd backend* to retrieve them and execute them.
+    - interactions between the xenopsd server and its backend occur
+    bidirectionnaly using the broker.
+
+## A diagram is worth a thousand words
+
+```mermaid
+
+flowchart TB
+
+    %% First we are starting by a XAPI client that is sending an XML-RPC request
+    client((Xapi client)) --|sends RPC XML request|-->
+        xapi_server{"`Dispatch RPC
+                    **api_server.ml**`"}
+    style client stroke:#CAFEEE,stroke-width:4px
+
+    %% XAPI Toolstack internals
+    subgraph "Xapi Toolstack"
+        style server stroke:#BAFA00,stroke-width:4px,stroke-dasharray: 5 5
+
+        xapi_server --|dispatch call (ie VM.pool_migrate)|--> server("`Auto generated using *IDL*
+                **server.ml**`")
+
+        server --|do_dispatch (ie VM.pool_migrate)|--> server_helpers["`server helpers
+        **server_helpers.ml**`"]
+
+        server_helpers --| call management (ie xapi_vm_migrate.ml)|--> vm_management["`VM Mgmt
+        like **xapi_vm_migrate.ml**`"]
```

Comment:
The call is first handled by the forwarding logic in `message_forwarding.ml`. This determines which host in the pool must execute the call, and forwards it to that host. That host then executes the local function in `xapi_vm_migrate.ml`.

Reply:
ok got it. I will fix it.